services/tenant-ui/frontend/src/store/utils/fetchItem.ts

```diff
@@ -0,0 +1,37 @@
+import { useTenantApi } from '../tenantApi';
+import { Ref } from 'vue';
+
+export async function fetchItem(url: string, id: string, dict: any, error: Ref<any>, loading: Ref<boolean>, params: any = {}) {
```

Comment:
don't store these items in a dict, just fetch and use on demand. not sure if you want to pass loading in here since it isn't being used and isn't loading the collection.